classes/form/CustomerForm.php

```diff
@@ -317,8 +317,14 @@ private function validateByModules()
                 // ToDo : replace Hook::exec with HookFinder, because we expect a specific class here
                 $validatedCustomerFormFields = Hook::exec('validateCustomerFormFields', ['fields' => $formFields], $moduleId, true);
 
-                if (is_array($validatedCustomerFormFields)) {
-                    $this->formFields = array_merge($this->formFields, $validatedCustomerFormFields);
+                if (!is_array($validatedCustomerFormFields)) {
+                    continue;
+                }
+
+                foreach ($validatedCustomerFormFields as $name => $field) {
+                    if ($field instanceof FormFieldCore) {
```

Comment:
Replace FormFieldCore by FormField

Reply:
It should work for the `FormField` as well

Reply:
`FormField` is supposed to extend `FormFieldCore` anyway so using the parent class for the check should also work, I don't think it's worth blocking the PR